algorithms_task_2/student/include/impl.h

```diff
@@ -9,4 +10,16 @@
  *
  * @return true if source starts from a prefix
  */
-bool startsWith(const std::string& source, const std::string& prefix);
+bool startsWith(const std::string& source, const std::string& prefix){
+    return std::equal(prefix.begin(),prefix.end(),source.begin()); 
+    // Complexity: prefix.begin()-prefix.end() applications of the predicate
+
+    // Alternatives:
+    // 1) return source.find(prefix)==0; 
+    //    Complexity: Unspecified, but generally up to linear in length()-pos times the length of the sequence to match (worst case)
+
+    // 2) return std::search(source.begin(),source.end(),prefix.begin(),prefix.end())==source.begin(); 
+    //    Complexity: At most S*N comparisons where S = std::distance(prefix.begin(),prefix.end()) and N = std::distance(source.begin(),source.end())
+
+    // Conclusion: std::equal is the best solution for complexity and for code readability 
```

Comment:
and what is the difference with mismatch?

Reply:
Maybe that in C++11 mismatch is less safer than search because it does not provide a check on the sizes and assume that the second range has at least many elements as the first

Reply:
same as std::equal
but mismatch will you give iterators, while equal - not